core/src/main/java/smile/validation/StratifiedCrossValidation.java

```diff
@@ -0,0 +1,143 @@
+/*******************************************************************************
+ * Copyright (c) 2010 Haifeng Li
+ *   
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *  
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *******************************************************************************/
+
+package smile.validation;
+
+/**
+ * An extension of Cross-validation for multi-class problems that adds an
+ * assertion that all partition keep a similar ratio between classes as the
+ * original data.
+ *
+ * @author Noam Segev
+ */

```

Comment:
Shall we add this to CrossValidation class with an optional parameter "boolean stratified" to control the behavior? The default value should be false so that we won't break existing code.